roles/openshift_web_console/tasks/install.yml

```diff
@@ -0,0 +1,79 @@
+---
+# Fact setting
+- name: Set default image variables based on deployment type
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ openshift_deployment_type | default(deployment_type) }}.yml"
+    - "default_images.yml"
+
+- name: Set openshift_web_console facts
+  set_fact:
+    openshift_web_console_prefix: "{{ openshift_web_console_prefix | default(__openshift_web_console_prefix) }}"
+    openshift_web_console_version: "{{ openshift_web_console_version | default(__openshift_web_console_version) }}"
+    openshift_web_console_image_name: "{{ openshift_web_console_image_name | default(__openshift_web_console_image_name) }}"
+    # Default the replica count to the number of masters.
+    openshift_web_console_replica_count: "{{ openshift_web_console_replica_count | default(groups.oo_masters_to_config | length) }}"
+
+- name: Ensure openshift-web-console project exists
+  oc_project:
+    name: openshift-web-console
+    state: present
+
+- name: Make temp directory for asset config files
+  command: mktemp -d /tmp/console-ansible-XXXXXX
+  register: mktemp
+  changed_when: False
+  become: no
+
+- name: Copy asset config template to temp directory
+  copy:
+    src: "{{ __console_files_location }}/{{ item }}"
+    dest: "{{ mktemp.stdout }}/{{ item }}"
+  with_items:
+    - "{{ __console_template_file }}"
+    - "{{ __console_config_file }}"
+
+- name: Update asset config properties
```

Comment:
I was advised to do it this way so that we can keep the same files for cluster up and the ansible install, following the pattern used for the template service broker. Cluster up can't use a jinja template.

We might be able to change the values as parameters in the template itself, though.

Cluster up changes are here: https://github.com/openshift/origin/pull/17575

cc @deads2k

Reply:
FWIW it's the same pattern used here: https://github.com/openshift/openshift-ansible/blob/master/roles/template_service_broker/tasks/install.yml#L26-L39

Reply:
> FWIW it's the same pattern used here

I appreciate you trying to keep things uniform.  We've been having some discussions around these upstream templates since that code merged.  Some people want to auto-sync their templates into our repos and apply changes via yedit.  I'm against that approach.

I appreciate the desire to have templates that are used across both projects.  But as it stands currently, that's not really the case if we have to create temporary files, edit them, oc template them, and then apply them.

I don't have a strong preference between modifying the templates upstream vs using a proper jinja template here.  But I don't want to yedit each of upstream's templates because it's going to be much harder to maintain IMO.

Reply:
> But I don't want to yedit each of upstream's templates because it's going to be much harder to maintain IMO.

Not necessarily disagreeing, but I want to make it clear we're not calling yedit on the template. We're calling yedit on a vanilla YAML config file and that value is passed as a variable to the template.

Reply:
@sdodson Can you weigh in on this one?  Not syncing or otherwise using the files that are maintained by the component owners in the repos they own that are used for integration and unit testing seems very risky.  The dependency isn't likely to ever go in the other direction, so not sync'ing seems like a recipe for drift.

I expect the default config map content and the default template content to be owned by the local owners in the repo they develop in.  I think we've better success at keeping those up to date than we have keeping other disjoint templates up to date.

Reply:
It sounds like we need better integration and regression testing across these components with openshift-ansible.